gnark2json/gnark2json.go

```diff
@@ -0,0 +1,78 @@
+package gnark2zkv
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+)
+
+// SaveToJSON saves a given data structure (v) to a JSON file at the specified filePath.
+// It returns an error if the operation fails.
+func SaveToJSON(filePath string, v interface{}) error {
+	// MarshalIndent generates a JSON string with indentation for better readability.
+	jsonData, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	// Write the JSON data to the specified file.
+	err = ioutil.WriteFile(filePath, jsonData, 0644)
```

Comment:
ioutil.WriteFile seems to be deprecated in favor of os.WriteFile ?

Reply:
Updated ioutil.WriteFile to os.WriteFile as recommended. Thank you for the suggestion!